### PR TITLE
Add support for Nuvoton numicro m48x SoC series

### PR DESCRIPTION
**Pull request by ssekar15**
This PR adds support for Nuvoton numicro m48x SoC series with development board numaker pfm-m487. Currently the west module dependency points to https://github.com/linumiz/hal_nuvoton.

This has dependency on
1. openocd support for M4 series in below PR
https://github.com/zephyrproject-rtos/openocd/pull/26
2. RFC for hal layer
https://github.com/zephyrproject-rtos/zephyr/issues/24703

Support:
Initial revision adds support only for UART in polling mode
Uses inbuilt Nor Flash 512, and default SRAM (160KB)
Clock defaults to 192MHz
WIP
Clock Control, GPIO and PINMUX support work is in progress and will be submitted in separate PR once the basic structure for Nuvoton is accepted

### CODE REVIEW

**Comment by ssekar15** [comment]
depends on 
https://github.com/zephyrproject-rtos/hal_nuvoton/pull/1

**Comment by ssekar15** [comment]
@carlescufi please have a look into PR

**Comment by carlescufi** [comment]
@ssekar15 could you get this PR passing CI so I can merge https://github.com/zephyrproject-rtos/hal_nuvoton/pull/1?

**Comment by ssekar15** [comment]
> @ssekar15 could you get this PR passing CI so I can merge [zephyrproject-rtos/hal_nuvoton#1](https://github.com/zephyrproject-rtos/hal_nuvoton/pull/1)?

done

File: west.yml [hunk 1]
```diff
@@ -61,6 +61,9 @@ manifest:
     - name: hal_openisa
       revision: 40d049f69c50b58ea20473bee14cf93f518bf262
       path: modules/hal/openisa
+    - name: hal_nuvoton
+      revision: pull/1/head
```
Comment (carlescufi) [inline west.yml:65]: Please replace with b4d31f33238713a568e23618845702fadd67386f

**Review by carlescufi — APPROVED** [review]

**Comment by carlescufi** [comment]
CC @MulinChao

File: west.yml (hunk 1 above)
Comment (ssekar15) [reply west.yml]: done

**Review by jettr — APPROVED** [review]

File: soc/arm/nuvoton/m48x/soc.h [hunk 2]
```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright (c) 2020 Linumiz
+ * Author: Saravanan Sekar <saravanan@linumiz.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _M48X_SOC_H_
```
Comment (stephanosio) [inline soc/arm/nuvoton/m48x/soc.h:8]: Should use a conflict-free identifier; also should not use identifiers that start with an underscore as per various coding standards.

```suggestion
#ifndef ZEPHYR_SOC_ARM_NUVOTON_M48X_SOC_H_
```

File: soc/arm/nuvoton/m48x/soc.c [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+	/* Set core clock as PLL_FOUT source */
+	CLK_SetCoreClock(FREQ_192MHZ);
+
+	/* Set both PCLK0 and PCLK1 as HCLK/2 */
+	CLK->PCLKDIV = (CLK_PCLKDIV_APB0DIV_DIV2 | CLK_PCLKDIV_APB1DIV_DIV2);
+
+	SystemCoreClockUpdate();
+
+	SYS_LockReg();
```
Comment (stephanosio) [inline soc/arm/nuvoton/m48x/soc.c:32]: Note that these functions are executed before the bss section is zeroed and the data sections are copied to the RAM.

Are you completely certain that none of these functions and the functions called by them make use of the pre-loaded data section contents or rely on the initialised bss section? If not, these should be called through `SYS_INIT` after `prep_c`.

Comment (ssekar15) [reply soc/arm/nuvoton/m48x/soc.c:32]: @stephanosio thanks and I aware of the hook z_platform_init is called before C init. non of these functions uses global/static variables and all of these are direct register address accessing.

File: soc/arm/nuvoton/m48x/soc.h (hunk 2 above)
Comment (ssekar15) [reply soc/arm/nuvoton/m48x/soc.h:8]: ok I will change

**Review by stephanosio — APPROVED** [review]

**Pull request merged**